src/vmm/tests/mock_seccomp/mod.rs

```diff
@@ -207,7 +204,7 @@ impl MockSeccomp {
             ]
             .into_iter()
             .collect(),
-            default_action: SeccompAction::Trap,
+            default_action: SeccompAction::Kill,
```

Comment:
Seccomp [documentation](http://man7.org/linux/man-pages/man2/seccomp.2.html) pertaining to `SECCOMP_RET_KILL_PROCESS` says that:
> The process terminates as though killed by a SIGSYS signal.
    Even if a signal handler has been registered for SIGSYS, the
    handler will be ignored in this case and the process always
    terminates.

Based on this doc, I can not follow the logic. The `mock_sigsys_handler` would have to be executed, and then the `SIGSYS_RECEIVED` boolean changed to `true`, for this test to work with success. However, the docs says that when receiving `SECCOMP_RET_KILL_PROCESS`, the  registered SIGSYS handler won't be executed.

Am I missing something?

Reply:
The doc says that the signal handler will be ignored even if present. I deleted it. The test now passes on asserting that the child's exit code is "death by `SIGSYS`".

The reason behind this change is that, with `SECCOMP_RET_TRAP` (as was before), the signal handler is **never** executed **when running under `kcov`**. The same happens in unit tests, where if `kcov` is detected (by relying on the fact that it forces CPU affinity to 1), the signal handler part is not verified.

Reply:
Oh, right. I've looked again on the diff, and missed some part of the changes. It makes sense now, thanks.